src/component/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ChangeEvent, useState, KeyboardEvent } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+interface TodoInfoProps {
+  todo: Todo;
+  deleteTodo: (id: number) => void;
+  updateTodoCompleted: (id: number, completed: boolean) => void;
+}
+
+export const TodoInfo: React.FC<TodoInfoProps> = ({
+  todo,
+  deleteTodo,
+  updateTodoCompleted,
+}) => {
+  const { id, title, completed } = todo;
+  const [isTitleEdit, setIsTitleEdit] = useState(false);
+  const [updatedTitle, setUpdatedTitle] = useState(title);
+
+  const handleDoubleClick = () => {
+    setIsTitleEdit(true);
+  };
+
+  const handleUpdateTitle = () => {
+    if (!updatedTitle.trim()) {
+      deleteTodo(id);
+    }
+
+    if (updatedTitle === title) {
+      setIsTitleEdit(false);
+
+      return;
+    }
+
+    setIsTitleEdit(false);
+  };
+
+  const handleChangeTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    setUpdatedTitle(event.target.value);
+  };
+
+  const cancelEditing = (event: KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setUpdatedTitle(title);
+    }
+  };
+
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      handleUpdateTitle();
+    }
+  };
+
+  const handleCheckboxChange = () => {
+    const updatedCompleted = !completed;
+
+    updateTodoCompleted(id, updatedCompleted);
+  };
+
+  return (
+    <div
+      key={id}
```

Comment:
Using 'key' prop on the root element of a component is unnecessary. Remove the 'key' prop from the 'div' element.